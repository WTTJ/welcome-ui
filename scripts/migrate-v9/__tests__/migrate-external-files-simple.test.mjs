import { readFileSync } from 'fs'
import { resolve } from 'path'

import { describe, expect, it } from 'vitest'

import { setupExternalMigrationTest } from '../helpers/test-setup.mjs'
import { migrateAll } from '../index.mjs'

describe('External Migration - Simple Styled Components', () => {
  // Don't perform cleanup for now so that we can inspect the migrated files
  const tempDir = setupExternalMigrationTest(
    'ExternalSimpleComponent-test',
    'ExternalSimpleComponent'
  )

  it('migrates simple styled components (styled.div, styled.button, etc.)', async () => {
    // Run migration
    await migrateAll(tempDir, { copyDir: false, interactive: false })

    // Read results
    const migratedComponent = readFileSync(resolve(tempDir, 'index.tsx'), 'utf8')
    const generatedScss = readFileSync(resolve(tempDir, 'styles.module.scss'), 'utf8')

    // Verify with snapshots
    expect(migratedComponent).toMatchSnapshot('simple-component.tsx')
    expect(generatedScss).toMatchSnapshot('simple-styles.module.scss')
  })
})
