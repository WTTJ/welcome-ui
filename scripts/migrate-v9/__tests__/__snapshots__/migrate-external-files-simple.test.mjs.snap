// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`External Migration - Simple Styled Components > migrates simple styled components (styled.div, styled.button, etc.) > simple-component.tsx 1`] = `
"import styles from './styles.module.scss'
import { classNames } from 'welcome-ui/utils'
const cx = classNames(styles)
export const SimpleComponent = () => {
  return (
    <div className={cx('container')}>
      <Text className={cx('header')} as="h1">
        Simple Header
      </Text>
      <section className={cx('content')}>
        <button className={cx('button')}>Click me</button>
      </section>
    </div>
  )
}
"
`;

exports[`External Migration - Simple Styled Components > migrates simple styled components (styled.div, styled.button, etc.) > simple-styles.module.scss 1`] = `
"@layer components {
  .container {
    display: flex;
    flex-direction: column;
    padding: var(--spacing-lg);
  }

  .header {
    color: var(--color-primary-500);
    margin-bottom: var(--spacing-md);
  }

  .content {
    flex: 1;
  }

  .button {
    background-color: var(--color-primary-500);
    color: white;
    padding: var(--spacing-sm) var(--spacing-md);
    border: none;
    border-radius: var(--spacing-sm);
    cursor: pointer;
  }
}"
`;
